### PR TITLE
fix(getEarnPositions): Only support mainnet networkIds in production and testnet networkIds in staging

### PR DESCRIPTION
**Pull request by jh2oman**
This change will make it easier for blockchain api to get all available earn positions.

### CODE REVIEW

**Review by finnian0826 — APPROVED** [review]
lgtm, should we also do something similar for `EARN_SUPPORTED_POSITION_IDS` since those are networkId dependent? having the arbitrum-sepolia pool in that list when on mainnet shouldn't be an issue, just a way to cleanup

File: src/config/index.ts [hunk 1]
```diff
@@ -64,6 +68,12 @@ export function getConfig(): Config {
       .string()
       .default('')
       .transform((val) => (val === '' ? [] : val.split(','))),
+    EARN_SUPPORTED_NETWORK_IDS: z
+      .string()
+      .default('')
+      .transform((val) =>
+        val === '' ? ([] as NetworkId[]) : (val.split(',') as NetworkId[]),
```
Comment (finnian0826) [inline src/config/index.ts:75]: Is there a reason for the empty string check here but not in prod?

Comment (jh2oman) [reply src/config/index.ts:75]: here the string can be optional and default to empty string. in prod its not optional so we don't need to handle the empty case

**Comment by jh2oman** [comment]
@finnian0826 I think that `EARN_SUPPORTED_POSITION_IDS` is fine where it is actually. I think of that as more some constants as opposed to per environment config. And now that the supported networks are per-environment there is no way that users could accidentally access sepolia from mainnet or vice versa.

**Pull request merged**